### DISCUSSION TITLE
How to Give name and label to Checkbox dynamically using React Hook Form useFieldArray

### DISCUSSION THREAD

**Discussion by irfan1452** [react-hook-form/react-hook-form#9962] [Q&A]
I want to Implement User Permission Form Using React Hook Form useFieldArray

I Have Implemented the Form [codesandbox ](https://codesandbox.io/s/funny-andras-4n69h4?file=/src/Permission/Permission.tsx) and it's ok but The Problem is the Displaying Label of the Check Box dynamically and assigning a name to the checkbox,
 I'm using Material UI
I Have the Following Interface For The Form
```
export interface UserPermission {
  controllerID:          string;
  controllerName:        string;
  userPermissionActions: UserPermissionAction[];
}

export interface UserPermissionAction {
  actionID:          string;
  actionName:        string;
  isAllowed:         boolean;
}
```
```
<RHFCheckbox
                    name={`test[${nestIndex}].userPermissionActions[${index}].actionID`}
                    label={`test[${nestIndex}].userPermissionActions[${index}].actionName`}
                  />
```

**Comment by Sajarin-M** [comment]
Because you are passing a file path as the label prop to `RHFCheckbox`, you can watch it and pass it as the checkbox label like
```js
function RHFCheckbox({ label }) {
  ...
  const labelValue = useWatch({ name: label });
  return (
    ...
      <FormControlLabel
        label={labelValue}
      />
    ...
  );
}
```
Additionally, it seems like you are trying to toggle the `isAllowed` field on checkbox click. However, with your current implementation, the `actionID` field is toggling. I think you need to change the name prop like
```diff
<RHFCheckbox
-  name={`test[${nestIndex}].userPermissionActions[${index}].actionID`}
+  name={`test[${nestIndex}].userPermissionActions[${index}].isAllowed`}
/>
```

**Reply by irfan1452** [reply]
Thanks, Sajarin
I did the following and its working now

```
 const nestedValues = useWatch({
    name: `test[${nestIndex}].userPermissionActions`
  });


<RHFCheckbox
                    name={`test[${nestIndex}].userPermissionActions[${index}].isAllowed`}
                    label={nestedValues[index].actionName}                   
                  />
```

**Reply by irfan1452** [reply]
But now the RHFCheckbox onChange Function not working
